Pretty.fs

```diff
@@ -37,10 +40,38 @@ let indent level = new string(' ', level * 4)
 /// Enters a new line at the given indent level.
 let lnIndent level = "\n" + indent level
 
+/// Helpers for turning a Doc into a Styled
+/// for syntax highlighting keywords, literals, identifiers and view syntax
+/// respectively
+let syntax d = Styled([Magenta], d)
+let syntaxLiteral d = Styled([Blue], d)
+let syntaxIdent d = Styled([Cyan], d)
+let syntaxView d = Styled([Yellow], d)
+
+/// Magic
+let stylise s d = 
+    let colCode =
+        function
+        | Black -> 0
+        | Red -> 1
+        | Green -> 2
+        | Yellow -> 3
+        | Blue -> 4
+        | Magenta -> 5
+        | Cyan -> 6 
+        | White -> 7
+
+    let code c = (30 + colCode c).ToString ()
```

Comment:
Changing this to eg. `sprintf "%d" (30 + colCode c)` removes the warning about copying.  I wonder if F# thinks that `ToString ()` can somehow mutate `(30 + colCode c)`.